src/tallies/tally_filter_zernike.F90

```diff
@@ -108,7 +123,74 @@ function text_label(this, bin) result(label)
         exit
       end if
     end do
-  end function text_label
+  end function text_label_zn
+
+!===============================================================================
+! ZernikeRadialFilter methods
+!===============================================================================
+
+  subroutine from_xml_zn_rad(this, node)
+    class(ZernikeRadialFilter), intent(inout) :: this
+    type(XMLNode), intent(in) :: node
+
+    integer :: n
+
+    ! Get center of cylinder and radius
+    call get_node_value(node, "x", this % x)
+    call get_node_value(node, "y", this % y)
+    call get_node_value(node, "r", this % r)
+
+    ! Get specified order
+    call get_node_value(node, "order", n)
+    this % order = n
+    this % n_bins = n/2 + 1
```

Comment:
There's an opportunity to cut down on redundant code here.  The only difference between `from_xml_zn` and `from_xml_zn_rad` is this `n_bins` calculation.  So let's break that small part into a separate procedure.

Make a new procedure called something like `calc_n_bins` which is a function that uses `this % order` to return the right number of bins.  Then you'll be able to do `this % n_bins = this % calc_n_bins()` and use only one `from_xml` procedure for both classes.

Reply:
It is indeed faster. 
Do you mean by adding a procedure like `calc_n_bins` in `ZernikeFilter`? Then I need the same subroutine in `ZernikeRadialFilter` as well. The redundancy is still there.

Reply:
You would be able to remove `from_xml_zn_rad` and use the same `from_xml` for both functions.  I was hoping to find other redundancies that could be removed, but it's tough to go beyond that so it's fine if you leave the separate `from_xml` procedures.

Reply:
Fixed this anyway. See if new commits work.